inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/experimental_detectron_roi_feature_extractor/experimental_detectron_roi_feature_extractor_kernel_ref.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "experimental_detectron_roi_feature_extractor_kernel_ref.h"
+#include <algorithm>
+#include <string>
+
+namespace kernel_selector {
+namespace {
+    ExperimentalDetectronROIFeatureExtractorRef::DispatchData SetDefault(const experimental_detectron_roi_feature_extractor_params& params) {
+        ExperimentalDetectronROIFeatureExtractorRef::DispatchData dispatch_data;
+
+        dispatch_data.gws[0] = params.output.LogicalSize();
+        dispatch_data.gws[1] = 1;
+        dispatch_data.gws[2] = 1;
+
+        dispatch_data.lws[0] = std::min(std::max(dispatch_data.gws[0], static_cast<size_t>(1)), static_cast<size_t>(32));
+        while (dispatch_data.gws[0] % dispatch_data.lws[0] != 0) {
+            --dispatch_data.lws[0];
+        }
+        dispatch_data.lws[1] = 1;
+        dispatch_data.lws[2] = 1;
```

Comment:
Such work group sizes may not be optimal in some cases, you can try something like:
```
dispatch_data.gws[0] = Align(params.output.LogicalSize(), 32);
dispatch_data.lws[0] = 32;
```
And check the range inside the kernel:
```
if (i >= OUTPUT_LENGTH)
    return;
```

Or even use something like this:
```
dispatchData.gws = {output.X().v, output.Y().v, output.Feature().v * output.Batch().v};
dims_by_gws = {{Tensor::DataChannelName::X},
              {Tensor::DataChannelName::Y},
              {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo, in_layout, out_layout, dims_by_gws);
```

Reply:
I've used second variant